### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by DenNegura**
Unit tests

### CODE REVIEW

File: src/test/java/com.endava.internship.collections/StudentMapTest.java [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class StudentMapTest {
+
+    public static List<Student> students;
```
Comment (rusnmaxim) [inline src/test/java/com.endava.internship.collections/StudentMapTest.java]: no need to be public here, rather private. Also if it was me I would not use method BeforeAll, just used here the List.of() and make it a constant

File: src/test/java/com.endava.internship.collections/StudentMapTest.java [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+    StudentMap map;
```
Comment (rusnmaxim) [inline src/test/java/com.endava.internship.collections/StudentMapTest.java]: private

File: src/test/java/com.endava.internship.collections/StudentMapTest.java [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+    @Test
+    public void whenInsertValue_mapIsNotEmpty() {
+        map.put(students.get(0), 10);
+
+        assertFalse(map.isEmpty());
+        assertEquals(1, map.size());
+    }
+
+    @Test
+    public void whenValueIsInput_mapReturnValue() {
```
Comment (rusnmaxim) [inline src/test/java/com.endava.internship.collections/StudentMapTest.java]: testGet

File: src/test/java/com.endava.internship.collections/StudentMapTest.java [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+        assertEquals(20, map.get(students.get(1)));
+        assertEquals(30, map.get(students.get(2)));
+        assertEquals(40, map.get(students.get(3)));
+        assertEquals(50, map.get(students.get(4)));
+        assertNull(map.get(null));
+        assertNull(map.get(students.get(5)));
+    }
+
+    @Test
+    public void whenValueInput_keyAndValueExist() {
```
Comment (rusnmaxim) [inline src/test/java/com.endava.internship.collections/StudentMapTest.java]: testContainsKeyAndValue

File: src/test/java/com.endava.internship.collections/StudentMapTest.java [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+        assertTrue(map.containsKey(students.get(0)));
+        assertTrue(map.containsValue(10));
+        assertTrue(map.containsKey(null));
+        assertTrue(map.containsValue(null));
+        assertFalse(map.containsKey(students.get(1)));
+        assertFalse(map.containsValue(20));
+    }
+
+    @Test
+    public void testFromMethodRemove() {
```
Comment (rusnmaxim) [inline src/test/java/com.endava.internship.collections/StudentMapTest.java]: testRemove

File: src/test/java/com.endava.internship.collections/StudentMapTest.java [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+        assertEquals(40, map.remove(students.get(3)));
+        assertEquals(50, map.remove(students.get(4)));
+        assertNull(map.remove(null));
+        assertNull(map.remove(students.get(5)));
+        assertNull(map.get(students.get(0)));
+        assertEquals(size - 6, map.size());
+    }
+
+    @Test
+    public void testFromMethodClear() {
```
Comment (rusnmaxim) [inline src/test/java/com.endava.internship.collections/StudentMapTest.java]: testClear

File: src/test/java/com.endava.internship.collections/StudentMapTest.java [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+        map.put(students.get(2), 30);
+        Set<Student> keys = map.keySet();
+
+        for(Student key : keys) {
+            assertTrue(map.containsKey(key));
+        }
+    }
+
+    @Test
+    public void testResize() {
```
Comment (rusnmaxim) [inline src/test/java/com.endava.internship.collections/StudentMapTest.java:198]: good job

File: src/test/java/com.endava.internship.collections/StudentMapTest.java [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+
+        int countOfResize = countResize(map.size(), 1, 0.5f);
+
+        System.setOut(out);
+        String str = log.toString();
+        int countOfResizeMap = str.split("\n").length;
+
+        assertEquals(countOfResize, countOfResizeMap);
+    }
+    public int countResize(int countStudents, int capacity, float load) {
```
Comment (rusnmaxim) [inline src/test/java/com.endava.internship.collections/StudentMapTest.java]: private static final

File: src/test/java/com.endava.internship.collections/StudentMapTest.java (hunk 1 above)
Comment (DenNegura) [reply src/test/java/com.endava.internship.collections/StudentMapTest.java]: I agree, but I don't know why List.of() doesn't work. I used Stream.of(...).collect(Collectors.toList())

File: src/test/java/com.endava.internship.collections/StudentMapTest.java (hunk 2 above)
Comment (DenNegura) [reply src/test/java/com.endava.internship.collections/StudentMapTest.java]: Fixed

File: src/test/java/com.endava.internship.collections/StudentMapTest.java (hunk 3 above)
Comment (DenNegura) [reply src/test/java/com.endava.internship.collections/StudentMapTest.java]: Renamed

File: src/test/java/com.endava.internship.collections/StudentMapTest.java (hunk 4 above)
Comment (DenNegura) [reply src/test/java/com.endava.internship.collections/StudentMapTest.java]: Renamed!

File: src/test/java/com.endava.internship.collections/StudentMapTest.java (hunk 5 above)
Comment (DenNegura) [reply src/test/java/com.endava.internship.collections/StudentMapTest.java]: last title removed. Rename

File: src/test/java/com.endava.internship.collections/StudentMapTest.java (hunk 6 above)
Comment (DenNegura) [reply src/test/java/com.endava.internship.collections/StudentMapTest.java]: last title cleared. Rename

File: src/test/java/com.endava.internship.collections/StudentMapTest.java (hunk 7 above)
Comment (DenNegura) [reply src/test/java/com.endava.internship.collections/StudentMapTest.java:198]: Thanks!

File: src/test/java/com.endava.internship.collections/StudentMapTest.java (hunk 8 above)
Comment (DenNegura) [reply src/test/java/com.endava.internship.collections/StudentMapTest.java]: fixed!

File: src/test/java/com.endava.internship.collections/StudentMapTest.java (hunk 1 above)
Comment (rusnmaxim) [reply src/test/java/com.endava.internship.collections/StudentMapTest.java]: as I see from the pom, your java is automaticaly configured as 1.8, but List.of() appeared in 1.9 or later

File: src/test/java/com.endava.internship.collections/StudentMapTest.java [hunk 9]
```diff
@@ -0,0 +1,231 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class StudentMapTest {
+
+    private final static List<Student> students = Stream.of(
```
Comment (rusnmaxim) [inline src/test/java/com.endava.internship.collections/StudentMapTest.java:23]: image
image
image
the way to fix it was this

**Review by rusnmaxim — APPROVED** [review]

File: src/test/java/com.endava.internship.collections/StudentMapTest.java (hunk 9 above)
Comment (DenNegura) [reply src/test/java/com.endava.internship.collections/StudentMapTest.java:23]: I got you. thanks

**Pull request merged**